src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -10476,14 +10477,27 @@ private ExpressionSyntax ParseTermWithoutPostfix(Precedence precedence)
                 case SyntaxKind.StackAllocKeyword:
                     return this.ParseStackAllocExpression();
                 case SyntaxKind.DelegateKeyword:
+                    if (this.IsPossibleLambdaExpression(precedence))
+                    {
+                        return this.ParseLambdaExpression();
+                    }
```

Comment:
i'm confused by this.  what case is this supporting?

---

hrmm... is this so that if we have a lmbda that returns a function pointer that we properly handle things? if so, can you please doc?